### PR TITLE
transformers support generation, trainer, tutorial, etc.

### PR DESCRIPTION
**Pull request by zhanghuiyao**
# What does this PR do?

Adds # (feature)
1. generation
2. trainer
3. some tutorials, readme, docs
4. llama3 8b infer/gen/finetune

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [x] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/transformers/llama/finetune_in_native_mindspore.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
+""" Llama 3 model fine-tuning script.
+    This script with default values fine-tunes a pretrained Meta Llama3 on the `Yelp/yelp_review_full` dataset,
+
+    Run with multiple cards, example as 8 cards:
+        msrun --bind_core=True --worker_num=8 --local_worker_num=8 --master_port=9000 --log_dir=outputs/parallel_logs \
+        python finetune_in_native_mindspore.py \
+          --model_path meta-llama/Meta-Llama-3-8B \
+          --dataset_path Yelp/yelp_review_full \
+          \
```
Comment (CaitinZhao) [inline examples/transformers/llama/finetune_in_native_mindspore.py:9]: delete

File: mindone/transformers/mindspore_adapter/amp.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                nn.GELU,
+                nn.Softmax,
+                nn.Sigmoid,
+                nn.MaxPool1d,
+                nn.MaxPool2d,
+                nn.MaxPool3d,
+                nn.AvgPool1d,
+                nn.AvgPool2d,
+                nn.AvgPool3d,
+                nn.CrossEntropyLoss,
```
Comment (SamitHuang) [inline mindone/transformers/mindspore_adapter/amp.py:51]: this hard-coded fp32 layers may not fit for all models

File: mindone/transformers/mindspore_adapter/attention.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            ops.matmul(query, key.swapaxes(-2, -1)) / (query.shape[-1] ** 0.5), axis=-1, dtype=ms.float32
+        ).astype(query.dtype)
+
+    out = ops.matmul(attn_weight, value)
+    out = out.astype(ori_dtype)
+
+    return out
+
+
+class FlashAttention2(nn.Cell):
```
Comment (SamitHuang) [inline mindone/transformers/mindspore_adapter/attention.py:36]: may illustrate it's just a wrapper, not the real FA2.

File: mindone/transformers/mindspore_adapter/adamw.py
Comment (SamitHuang) [inline mindone/transformers/mindspore_adapter/adamw.py:1]: what about using mint adamw?

File: mindone/transformers/mindspore_adapter/recompute.py
Comment (SamitHuang) [inline mindone/transformers/mindspore_adapter/recompute.py:1]: not necessary for pynative?

File: mindone/transformers/README.md
Comment (SamitHuang) [inline mindone/transformers/README.md:1]: Can list the supported features compared to torch? For example, `beam_search` for generation, 8-bit quantization for memory reduction, are quite commonly used.

Comment (SamitHuang) [inline mindone/transformers/README.md:1]: better claim which ms version and mode (graph/pynative) are mainly tested. If both graph and pynative mode are supported, do both of them guarantee good accuracy? or just both runnable?

File: mindone/transformers/mindspore_adapter/attention.py (hunk 3 above)
Comment (zhanghuiyao) [reply mindone/transformers/mindspore_adapter/attention.py:36]: i heard before that it is really fa2

File: mindone/transformers/mindspore_adapter/recompute.py
Comment (zhanghuiyao) [reply mindone/transformers/mindspore_adapter/recompute.py:1]: may need

File: examples/transformers/llama/finetune_in_native_mindspore.py (hunk 1 above)
Comment (zhanghuiyao) [reply examples/transformers/llama/finetune_in_native_mindspore.py:9]: ok

File: mindone/transformers/README.md
Comment (zhanghuiyao) [reply mindone/transformers/README.md:1]: currently only the most basic `sample` method for generation is supported to provide for MLLMs to use, and do not support any quantification.

Comment (zhanghuiyao) [reply mindone/transformers/README.md:1]: the full interface will be provided in the subsequent version 4.46.2, including `beam_search` for generation

Comment (zhanghuiyao) [reply mindone/transformers/README.md:1]: the he newly added features were validated on ms2.3.1, but some existing interfaces may not be supported, this will require complete validation before providing a confirmed support version to the public.

Comment (zhanghuiyao) [reply mindone/transformers/README.md:1]: for infer/generate, it can have good accuracy, and for training, it is currently only in a runnable state.

**Review by SamitHuang — APPROVED** [review]

**Review by vigo999 — APPROVED** [review]
